src/ticpos.jl

```diff
@@ -0,0 +1,117 @@
+# This file is a part of AstroLib.jl. License is MIT "Expat".
+
+function _ticpos{T<:AbstractFloat}(deglen::T, pixlen::T, ticsize::T)
+    minpix = deglen*60/pixlen
+    incr = minpix*ticsize
+
+    if incr < 0
+        sgn = -1
+    else
+        sgn = 1
+    end
+
+    incr = abs(incr)
+
+    if incr >= 30
+        units = "Degrees"
+    elseif incr > 0.5
+        units = "Arc Minutes"
+    else
+        units = "Arc Seconds"
+    end
+
+    if incr >= 120
+        incr = 4
```

Comment:
Mind type-stability!  ;-)  Above, `incr` is an `AbstractFloat`, here you assign it an `Integer` value, causing instability.  You can either introduce a new variable (`incr1` or whatever) to which assign this integers, or keep `incr` but cast the integers to `T` (e.g., with `T(4)` in this line).

It really takes some exercise to do this right, I often find type-instabilities in my old code (last time happened to me just today)